### ISSUE TITLE
[Custom runtime] add auth 

### ISSUE THREAD

**Issue by vincentsarago** [developmentseed/eoAPI#3]
use Cognito 

https://github.com/tokusumi/fastapi-cloudauth

**Comment by zacharyDez** [comment]
@vincentsarago how should we think about this with the deployments across multiple cloud providers? Should we just leave it to the end-user to configure authentication based on their context?

**Comment by vincentsarago** [comment]
> Should we just leave it to the end-user to configure authentication based on their context?

I guess 🤷‍♂️ 

It would be nice to have it implemented in our different demo services still

**Comment by zacharyDez** [comment]
Yes, it came up during a call with some other external developers as being a key need that's unresolved for them...

**Comment by alukach** [comment]
A quick dump of thoughts on the topic of auth in reusable FastAPI systems:

On the VEDA STAC Ingestor, we have the maintainer provide the JWKS URL of whatever RS256-compliant auth provider you use (e.g. Cognito, Auth0) ([1]).  This was done with the intention of the codebase being agnostic & flexible regarding auth setups.  However, it comes at the cost of a lot of finer-grained controls.  The STAC Ingestor basically asks "is this a valid token?" ([2]) and nothing more.  It seems like [fastapi-cloudauth](https://github.com/tokusumi/fastapi-cloudauth) is a bit more intelligent in terms of validating provider-dependent details in the auth scope ([3], [4]), which most likely more secure. Additionally, it adds tooling for custom verification of scopes/claims, which seems useful for anything but the most basic auth requirements.

On STAC-FastAPI, we added a convenience function to allow implementors to add custom dependencies (e.g. auth requirements) to endpoints on an existing (ie imported) FastAPI app ([5]).  It feels a bit kludgey, but I'm not sure what a better solution would be for extending a FastAPI app with endpoints defined in files out of the control of the implementor.

One possible path forward would be to:

* Allow user to optionally provide JWKS url for situations where basic auth is appropriate and the implementor is using eoAPI out of the box
* Add & document/demonstrate a convenience method to make adding dependencies like those created via `fastapi-cloudauth`

[1]: https://github.com/NASA-IMPACT/veda-stac-ingestor/blob/552919e93187479b38d236ce1cd3b1e5433a9a8b/api/src/config.py#L15-L17
[2]: https://github.com/NASA-IMPACT/veda-stac-ingestor/blob/552919e93187479b38d236ce1cd3b1e5433a9a8b/api/src/auth.py#L25-L65
[3]: https://github.com/tokusumi/fastapi-cloudauth/blob/594153706391258d80590a31e31666f260519a83/fastapi_cloudauth/auth0.py#L92-L110
[4]: https://github.com/tokusumi/fastapi-cloudauth/blob/master/fastapi_cloudauth/cognito.py#L88-L103
[5]: https://github.com/stac-utils/stac-fastapi/pull/295

**Comment by vincentsarago** [comment]
eoAPI runtimes are `demo` applications. We should use them to show what we can do, so if we feel we should implement auth I'm totally 👍
